pkg/controller/deactivation/deactivation_controller_test.go

```diff
@@ -151,43 +151,123 @@ func TestReconcile(t *testing.T) {
+			mur := murtest.NewMasterUserRecord(t, username, murtest.Account("cluster1", *basicTier),
+				murtest.ProvisionedMur(murProvisionedTime), murtest.UserIDFromUserSignup(userSignupFoobar))
 			mur.Labels[toolchainv1alpha1.MasterUserRecordOwnerLabelKey] = userSignupFoobar.Name
-			userSignupFoobar.Status.Conditions = append(userSignupFoobar.Status.Conditions, toolchainv1alpha1.Condition{
-				Type:               toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
-				Status:             v1.ConditionTrue,
-				LastTransitionTime: metav1.Time{Time: time.Now().Add(time.Duration(preDeactivationNotificationDays*24*-1) * time.Hour)},
-				Reason:             toolchainv1alpha1.UserSignupDeactivatingNotificationCRCreatedReason,
-			})
+
 			r, req, cl := prepareReconcile(t, mur.Name, basicTier, mur, userSignupFoobar, config)
 			// when
 			res, err := r.Reconcile(req)
 			// then
 			require.NoError(t, err)
-			require.False(t, res.Requeue, "requeue should not be set")
-			require.True(t, res.RequeueAfter == 0, "requeueAfter should not be set")
-			assertThatUserSignupDeactivated(t, cl, username, true)
-			AssertMetricsCounterEquals(t, 1, metrics.UserSignupAutoDeactivatedTotal)
+			require.True(t, res.Requeue)
+			require.Equal(t, time.Duration(0)*time.Second, res.RequeueAfter)
+
+			// Reload the userSignup
+			require.NoError(t, cl.Get(context.TODO(), types.NamespacedName{Name: userSignupFoobar.Name, Namespace: operatorNamespace}, userSignupFoobar))
+			require.True(t, states.Deactivating(userSignupFoobar))
+			require.False(t, userSignupFoobar.Spec.Deactivated)
 
-			t.Run("usersignup already deactivated", func(t *testing.T) {
-				// additional reconciles should find the usersignup is already deactivated
+			t.Run("reconciliation should be requeued when notification not yet sent", func(t *testing.T) {
+				r, req, cl := prepareReconcile(t, mur.Name, basicTier, mur, userSignupFoobar, config)
+				// when
 				res, err := r.Reconcile(req)
 				// then
 				require.NoError(t, err)
-				require.False(t, res.Requeue, "requeue should not be set")
-				require.True(t, res.RequeueAfter == 0, "requeueAfter should not be set")
+				require.True(t, res.Requeue)
+				require.Equal(t, time.Duration(0)*time.Second, res.RequeueAfter)
+
+				// Reload the userSignup
+				require.NoError(t, cl.Get(context.TODO(), types.NamespacedName{Name: userSignupFoobar.Name, Namespace: operatorNamespace}, userSignupFoobar))
+
+				// deactivating state should still be true
+				require.True(t, states.Deactivating(userSignupFoobar))
+
+				// deactivated state should still be false
+				require.False(t, userSignupFoobar.Spec.Deactivated)
+
+				t.Run("usersignup requeued after deactivating notification created for user", func(t *testing.T) {
+					// Set the notification status condition as sent
+					userSignupFoobar.Status.Conditions = []toolchainv1alpha1.Condition{
```

Comment:
Shouldn't this condition be already true for `userSignupFoobar` set by the parent test?

Reply:
As discussed, this is set by the UserSignup controller.